src/mame/konami/thunderx.cpp

```diff
@@ -441,30 +433,27 @@ void thunderx_state::calculate_collisions()
 	// hit mask is 40 to set bit on object 0 and object 1
 	// hit mask is 20 to set bit on object 1 only
 
-	Y0 = m_pmcram[0];
-	Y0 = (Y0 << 8) + m_pmcram[1];
-	Y0 = (Y0 - 15) / 5;
-	Y1 = (m_pmcram[2] - 15) / 5;
+	const int e0 = (m_pmcram[0]<<8) | m_pmcram[1];
+	const int e1 = m_pmcram[2];
```

Comment:
Note that this actually does change behaviour – previously it divided the Y1 value by 5 and then multiplied by 5 again later.  If the actual value was not a multiple of 5, the remainder would be truncated.  That’s no longer the case.  It now uses the actual value from RAM directly.

Reply:
Yes, but data is accessed in groups of 5 later on. The new code makes more sense and is more parsimonius. I have not been able to find any problem in the gameplay after the change.

I do not know why the original author ended up writting those divisions and multiplications. They might be some leftover from all the guess work needed to emulate the chip or maybe he was trying to make sense of the _program_ bytes written by the CPU.